src/commands.rs

```diff
@@ -3867,61 +3867,89 @@ fn cmd_git_push(
-        } else {
-            writeln!(
-                ui,
-                "Branch {}@{} already matches {}",
-                branch_name, &args.remote, branch_name
-            )?;
+    if !args.branch.is_empty() {
+        for branch_name in &args.branch {
+            if let Some(update) = branch_updates_for_push(
+                workspace_command.repo().as_repo_ref(),
+                &args.remote,
+                branch_name,
+            )? {
+                branch_updates.push((branch_name.clone(), update));
+            } else {
+                writeln!(
+                    ui,
+                    "Branch {}@{} already matches {}",
+                    branch_name, &args.remote, branch_name
+                )?;
+            }
         }
+        let branch_description = match &args.branch[..] {
+            [only_branch] => format!("branch {only_branch}"),
+            [.., last_branch] => {
+                format!(
+                    "branches {} and {last_branch}",
+                    &args.branch[..args.branch.len() - 1].join(", ")
+                )
+            }
+            _ => panic!("Assumed there is at least one branch"),
+        };
         tx = workspace_command.start_transaction(&format!(
-            "push branch {branch_name} to git remote {}",
+            "push {branch_description} to git remote {}",
             &args.remote
         ));
-    } else if let Some(change_str) = &args.change {
-        let commit = workspace_command.resolve_single_rev(change_str)?;
-        let branch_name = format!(
-            "{}{}",
-            ui.settings().push_branch_prefix(),
-            commit.change_id().hex()
-        );
-        if workspace_command
-            .repo()
-            .view()
-            .get_local_branch(&branch_name)
-            .is_none()
-        {
-            writeln!(
-                ui,
-                "Creating branch {} for revision {}",
-                branch_name,
-                change_str.deref()
-            )?;
+    } else if !args.change.is_empty() {
+        let (mut commits, mut commits_str) = (vec![], vec![]);
+        for change_str in &args.change {
+            let commit = workspace_command.resolve_single_rev(change_str)?;
+            commits_str.push(commit.change_id().hex().clone());
+            commits.push(commit);
         }
+        let change_description = match &commits_str[..] {
+            [only_change] => format!("change {only_change}"),
+            [.., last_change] => {
+                format!(
+                    "changees {} and {last_change}",
+                    commits_str[..commits_str.len() - 1].join(", ")
+                )
+            }
+            _ => panic!("Assumed there is at least one change"),
+        };
         tx = workspace_command.start_transaction(&format!(
-            "push change {} to git remote {}",
-            commit.change_id().hex(),
+            "push {change_description} to git remote {}",
```

Comment:
Same here. 

If `join(", ")` is good enough, or extracted helper function to build singular/plural description, we won't need to collect `commits_str` separately.

Reply:
OK, let's go with join(", ") for now.